astropy/utils/iers/iers.py

```diff
@@ -775,7 +818,7 @@ def _substitute_iers_b(cls, table):
         Here, we use the bundled astropy IERS-B table to overwrite the values
         in the downloaded IERS-A table.
         """
-        iers_b = IERS_B.open()
+        iers_b = IERS_B.open(IERS_B_FILE)
```

Comment:
It turns out this makes testing extremely frustrating. The astropy cache provides a kind of shared global state that makes systematic testing very difficult - I finally gave up and did this when the previous approach gave me unpredictable results. (Note that PR #9182 provides a solution to this particular problem.) I think the solution is actually a larger structural change.

Reply:
I think with adequate documentation this could be made to work: 

* If the user wants a specific table version, use IERS_B.open("iers_b.table")
* If the user wants the built-in version, use IERS_B.open(IERS_B_FILE)
* If the user wants the cached version, falling back to IERS_B_FILE or raising an error for IERS_A, they can just do IERS_B.open()
* If the user wants auto-updating, use IERS_Auto.open() (which starts with whatever is in cache).

This works for earth_rotation_table as well as direct use of the tables.

It's still different from now, where using no-argument open gets you whatever is stored in the class-wide cls.iers_table. But I can make that change fairly easily.

Reply:
Agreed that from a user perspective, with the new `earth_rotation_table`, it is possible to select whatever one wants. Here, it is about the behaviour of `IERS_Auto` - where does it gets its IERS-B values. Choices are:
- Use `IERS-A` IERS-B values (but those are not quite OK);
- Use bundled IERS-B file (but can be out of date; would happen with your change);
- Use whatever the current default is - bundled until something specific downloaded prior to `IERS_Auto()` (current behaviour; note: currently, any IERS_B_URL downloaded in a previous session is ignored);
- Use most current already downloaded version (i.e., as current, but do not ignore previously downloaded IERS-B);
- Ensure IERS-B is kept up to date in the same way IERS-A is.

For this PR, my tendency would be to document the present state...

Reply:
Is this now resolved?